secure/storage/src/storage.rs

```diff
@@ -47,33 +50,137 @@ pub trait Storage: Send + Sync {
     ///
     /// This default implementation is useful for storage engines that only implement the simple
     /// Storage interface. More complex engines (e.g., those that support native crypto key
-    /// generation), should override this implementation.
+    /// generation and rotation), should override this implementation.
     fn generate_new_ed25519_key_pair(
         &mut self,
         key_pair_name: &str,
         policy: &Policy,
     ) -> Result<Ed25519PublicKey, Error> {
-        let mut seed_rng = OsRng::new().map_err(|e| Error::EntropyError(e.to_string()))?;
-        let mut rng = rand::rngs::StdRng::from_seed(seed_rng.gen());
-        let private_key = Ed25519PrivateKey::generate_for_testing(&mut rng);
-        let public_key = private_key.public_key();
-        self.create(key_pair_name, Value::Ed25519PrivateKey(private_key), policy)
-            .map(|_| public_key)
+        // Generate and store the new named key pair
+        let (private_key, public_key) = new_ed25519_key_pair()?;
+        self.create(key_pair_name, Value::Ed25519PrivateKey(private_key), policy)?;
+
+        // Set the previous key pair version to be the newly generated key pair. This is useful so
+        // that we can also set the appropriate policy permissions on the previous key pair version
+        // now, and not have to do it later on a rotation.
+        self.create(
+            &get_previous_version_name(key_pair_name),
+            Value::Ed25519PrivateKey(self.get_private_key_for_name(key_pair_name)?),
+            policy,
+        )
+        .map(|_| public_key)
```

Comment:
why perform the map though... why not just ? and Ok(public_key). I think that's more intuitive, but I don't know which is more rusty.

Reply:
Yeah, agreed! I'm not sure which is more rusty.. but the simple ? and Ok(public_key) seems easier to reason about. Done, thanks!